src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -36,6 +38,22 @@ use crate::state_backend::ManagerWrite;
 use crate::state_backend::Ref;
 use crate::struct_layout;
 
+/// Errors relating to handling signals
+#[derive(Debug, Eq, thiserror::Error, PartialEq)]
+pub enum SignalError {
+    Memory(#[from] BadMemoryAccess),
+    BadContext,
+}
+
+impl Display for SignalError {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        match self {
+            Self::Memory(e) => e.fmt(f),
+            Self::BadContext => write!(f, "BadContext"),
+        }
+    }
+}
```

Comment:
```suggestion
    #[error("Bad signal context")]
    BadContext,
}
```

Have a look at the [documentation of `thiserror`](https://docs.rs/thiserror/latest/thiserror/#details). You can use the `#[error]` attribute to generate the `Display` implementation.